internal/clonepool/clonepool.go

```diff
@@ -225,16 +241,30 @@ func (c *ClonePool) Claim(timeout time.Duration) (zdap.PublicClone, error) {
 	}
 	expires := time.Now().Add(timeout)
 	err = c.cloneContext.Z.SetUserProperty(*claim.Dataset, zfs.PropExpires, expires.Format(zfs.TimestampFormat))
+	c.triggerGCAfterDelay(timeout)
 	if err != nil {
 		return zdap.PublicClone{}, err
 	}
 	c.ClonesAvailable--
+	err = c.cloneContext.Z.SetUserProperty(*claim.Dataset, zfs.PropOwner, owner)
+	if err != nil {
+		return zdap.PublicClone{}, err
+	}
 
 	claim.APIPort = c.cloneContext.ApiPort
 	claim.Server = c.cloneContext.NetworkAddress
+	c.TriggerGC()
 	return *claim, nil
 }
 
+func (c *ClonePool) triggerGCAfterDelay(delay time.Duration) {
+	timer := time.NewTimer(delay)
+	go func() {
+		<-timer.C
```

Comment:
I think a simple `time.Sleep(delay)` would be better here.

Optionally only create new GO-routines for shorter delays, like <30min or something like that, if we have ttl/timeouts that are longer then that, maybe we could just handle them by the hourly action trigger instead?

Reply:
I think for now let's keep the delay as is even for longer durations since the timing can still be quite off compared to the hourly run. I'm guessing there isn't a very large performance overhead from sleeping goroutines?